AndroidAnnotations/androidannotations-core/androidannotations/src/main/java/org/androidannotations/internal/core/handler/ExtraHandler.java

```diff
@@ -72,12 +89,34 @@ public void process(Element element, HasExtras holder) {
 		}
 
 		JFieldVar extraKeyStaticField = createStaticExtraField(holder, extraKey, fieldName);
-		injectExtraInComponent(element, holder, extraKeyStaticField, fieldName);
-
 		if (holder instanceof HasIntentBuilder) {
 			String docComment = getProcessingEnvironment().getElementUtils().getDocComment(element);
-			createIntentInjectionMethod(element, (HasIntentBuilder) holder, extraKeyStaticField, fieldName, docComment);
+			createIntentInjectionMethod(param, (HasIntentBuilder) holder, extraKeyStaticField, fieldName, docComment);
 		}
+
+		JMethod injectExtrasMethod = holder.getInjectExtrasMethod();
+		JVar extras = holder.getInjectExtras();
+
+		DeclaredType enclosingClassType;
+		if (param.getKind() == ElementKind.PARAMETER) {
```

Comment:
it does not provide the enclosing element and never did. the `enclosingElement` here is the enhanced class.

what provided is, is `element` and `param` where `element` is always the annotated element (the field, method or one of the arguments) and `param` is either the annotated element (for field and or arguments in a method with mutliple arguments) or the first/only argument for method injection where the the annotation is on the method.